src/zenml/container_registries/base_container_registry.py

```diff
@@ -26,9 +26,13 @@ class BaseContainerRegistry(StackComponent):
 
     Attributes:
         uri: The URI of the container registry.
+        username: Username to authenticate with the container registry.
+        password: Password to authenticate with the container registry.
     """
 
     uri: str
+    username: Optional[str] = None
+    password: Optional[str] = None
```

Comment:
Shouldnt this be a secret

Reply:
I actually had some bigger plans for secrets, but probably won't make it into this PR: String attributes on a stack component could be either
- The actual value
- Referring to a secret (e.g. "$secret_name"), in which case the value would be replaced by the secret value when needed

I still think we should have this, but as it won't make it into this PR I'll include a secret in addition to the raw username and password for the container registries. (Why in addition instead of replacing them: Some environments might not have a dedicated secret manager available and passing the raw values would be the only solution)